src/python/py/models/builder.py

```diff
@@ -1689,61 +1694,73 @@ def make_attention_mask_reformatting_for_mha(self):
         # Make nodes for the attention mask subgraphs that reformat the
         # 2D attention mask (B, S) to 4D causal attention mask (B, N, S, T)
```

Comment:
I do not under stand this. I think the 2D mask is (B, T), and MHA supports 2D mask, shall we use that directly instead of converting to 4D in onnx graph?